### PR TITLE
Adds support for sonarqube output

### PR DESCRIPTION
**Pull request by adamnengland**
This PR adds support for the sonarqube [Generic Issue Import Format](https://docs.sonarqube.org/latest/analysis/generic-issue/)

In general, the change just requires adding `sonar` as an output format, implementing `Brakeman::Report::Sonar < Brakeman::Report::Base` and testing.

Has been successfully tested with sonarqube 8.4.1 (current production version).

### CODE REVIEW

File: lib/brakeman/report/report_sonar.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      "CRITICAL"
+    elsif confidence == 1
+      "MAJOR"
+    else
+      "MINOR"
+    end
+  end
+
+  def file_path(warning)
+    if tracker.options[:path_prefix]
```
Comment (presidentbeef) [inline lib/brakeman/report/report_sonar.rb:39]: Where did this come from? Unused, maybe?

**Comment by presidentbeef** [comment]
@adamnengland Can you take a look at my comment above and also merge in `main` since SARIF support went in? Thanks!

**Comment by adamnengland** [comment]
@presidentbeef Working on it this week.  Please keep open for a few more days.

**Comment by presidentbeef** [comment]
@adamnengland No problem, thanks for the update!

File: lib/brakeman/report/report_sonar.rb (hunk 1 above)
Comment (adamnengland) [reply lib/brakeman/report/report_sonar.rb]: I had emulated this from code_climate format, but it turns out I don't need the complexity here.  Dropped back to the file on the warning itself, and works fine.

**Comment by adamnengland** [comment]
@presidentbeef Merged together with the serif outputter, and cleaned up a few issues.  Let me know if you have other questions in your review.

File: test/tests/sonar_output.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    @@sonar ||= JSON.parse(Brakeman.run("#{TEST_PATH}/apps/rails3.2").report.to_sonar)
+  end
+
+  def test_for_expected_keys
+    assert (@@sonar.keys - ["issues"]).empty?
+  end
+
+  def test_for_issues_keys
+    issues_keys = ["engineId", "ruleId", "severity", "type", "primaryLocation", "effortMinutes"]
+    puts @@sonar["issues"][0].keys
```
Comment (presidentbeef) [inline test/tests/sonar_output.rb:15]: Remove?

File: test/tests/sonar_output.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    assert (@@sonar["issues"][0].keys - issues_keys).empty?
```
Comment (presidentbeef) [inline test/tests/sonar_output.rb:16]: How about asserting this for each issue?

**Review by presidentbeef — CHANGES_REQUESTED** [review]

File: lib/brakeman/report/report_sonar.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
+class Brakeman::Report::Sonar < Brakeman::Report::Base
+  def generate_report
+    return {
+      issues: all_warnings.map { |warning| issue_json(warning) }
+    }.to_json
```
Comment (presidentbeef) [inline lib/brakeman/report/report_sonar.rb:5]: The JSON report uses `JSON.pretty_generate` instead of `to_json`, which makes the output nicely formatted.

**Comment by adamnengland** [comment]
@presidentbeef I've addressed the issues mentioned above.  Appreciate the feedback.

**Comment by presidentbeef** [comment]
Thanks @adamnengland!

**Comment by presidentbeef** [comment]
Thanks @adamnengland!

**Pull request merged**